src/Controllers/UserController.cs

```diff
@@ -1,25 +1,60 @@
 
 using Coffee_Shop_App.src.Abstractions;
-using Coffee_Shop_App.src.Services;
+using Coffee_Shop_App.src.Entities;
 using Microsoft.AspNetCore.Mvc;
 
 namespace Coffee_Shop_App.src.Controller;
 
 
+public class UserController : BaseController //the inheritance to get the routing
+{
+
+    private IUserService? _userService;
+    public UserController(IUserService userService)
+    {
+        _userService = userService;
+    }
 
-[ApiController]
-[Route("api/v1/[controller]")]
 
-public class UserController 
-{
 
+    [HttpGet] //import the ASP.NetCore package
+    public List<User>? findAll()
+    {
+        return _userService.FindAll(); //access to users's data
+    }
+
+
+    [HttpPost] //POST, PUT, or PATCH use fromBody
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    public ActionResult<User> CreateOne([FromBody] User user)
+    {
+        if (user is not null)
+        {
+
+            _userService.CreateOne(user); //sendin request to service
+            return CreatedAtAction(nameof(CreateOne), user); //return value in ActionResult
 
-    [HttpGet]
-    public string Hello()
+        }
+        return BadRequest(); //built-in method for validation
+    }
+
+
+    [HttpGet("{userId}")]
+    public User? findOne(string userId)
     {
-        UserService helloService = new ();
-        return helloService.SayHi();
+
+        return _userService.findOne(userId);
     }
 
-    
-}
\ No newline at end of file
+
+     [HttpGet("userEmail")]
+    public User? findOneByEmail(string userEmail)
+    {
+
+        return _userService.findOne(userEmail);
```

Comment:
here you have to call `findOneByEmail()` method from the `_userService` instead of `findOne()`